backend/src/ml_space_lambda/dataset/lambda_functions.py

```diff
@@ -107,6 +107,31 @@ def edit(event, context):
             raise Exception("Dataset description is over the max length of 254 characters.")
         if dataset_description_regex.search(body["description"]):
             raise Exception("Dataset description contains invalid character.")
+    if dataset.type == DatasetType.GROUP:
+        # get the new list of groups that have this dataset shared with them.
+        # this list may be adding or removing existing groups from this dataset
+        new_groups_list = body.get("groups", [])
+        # get the current list of groups with access to this dataset so we can determine which groups were added/removed
+        groups = group_dataset_dao.get_groups_for_dataset(dataset_name)
+        old_group_names = []
+        group_difference = []
+
+        # Check for existing groups that are no longer in the group list
```

Comment:
I think we cold be more efficient here by looping through by merging the new_groups_list and the groups list without dups and then looping through that once vs having two different loops. 

We can then do an if/else if to add/remove/ignore

Reply:
I was thinking through that earlier too... I could loop through both lists to create a list/set of the shared values, then loop again to determine what was added or removed... but this was easier and it seemed like it'd be the same time complexity overall.